gpytorch/utils/contour_integral_quad.py

```diff
@@ -0,0 +1,137 @@
+import math
+import warnings
+
+import torch
+
+from .broadcasting import _mul_broadcast_shape
+from .linear_cg import linear_cg
+from .minres import minres
+from .warnings import NumericalWarning
+
+
+def contour_integral_quad(
+    lazy_tensor,
+    rhs,
+    inverse=False,
+    weights=None,
+    shifts=None,
+    max_lanczos_iter=20,
+    num_contour_quadrature=7,
+    shift_offset=0,
+):
+    r"""
+    Performs :math:`\mathbf K^{1/2} \mathbf b` or `\mathbf K^{-1/2} \mathbf b`
+    using contour integral quadrature.
+
+    .. note::
+        Right now this only works for non-batch matrices
```

Comment:
will this fail gracefully?

Reply:
Oops, that note actually isn't even true anymore